### PR TITLE
add WxWork (企业微信) provider

### PR DESCRIPTION
**Pull request by keller31**
This is an WxWork provider has been tested
Hope to help those in need

### CODE REVIEW

**Comment by atymic** [comment]
Can you explain what wxwork is please?

**Comment by keller31** [comment]
> Can you explain what wxwork is please?

wxwork (WeCom's name in China)

WeCom is Tencent Inc a communication platform for enterprises that includes convenient communication and office automation tools. Similar to Weixin and WeChat, it offers smooth communication experience, as well as various OA applications. In addition, WeCom is integrated with Weixin, providing unmatched connectivy that facilitates smart management, smart ecosystem and smart services for enterprises.
https://work.weixin.qq.com/

File: src/WxWork/Provider.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+     * Unique Provider Identifier.
+     */
+    public const IDENTIFIER = 'WEIXINWORK';
+
+    protected $scopes = ['snsapi_base'];
+
+    protected $userid;
+
+    /**
+     * @param \SocialiteProviders\Manager\Contracts\OAuth1\ProviderInterface|\SocialiteProviders\Manager\Contracts\OAuth2\ProviderInterface $config
```
Comment (atymic) [inline src/WxWork/Provider.php:24]: This should be one or the other.

File: src/WxWork/Provider.php [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+     */
+    public function setConfig(ConfigInterface $config)
+    {
+        $config = config('services.weixinwork');
+
+        $this->config = $config;
```
Comment (atymic) [inline src/WxWork/Provider.php:30]: You should use `additionalConfigKeys` instead of setting this directly

File: src/WxWork/Provider.php [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+    /**
+     * {@inheritdoc}.
+     */
+    protected function getAuthUrl($state)
+    {
+        $isWxWork = Str::is('*wxwork*', request()->userAgent());
+
+        if ($isWxWork) {
+            //企业微信内部静默授权
```
Comment (atymic) [inline src/WxWork/Provider.php:47]: English comment instead or also here please

File: src/WxWork/Provider.php [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}.
+     */
+    protected function getAuthUrl($state)
+    {
+        $isWxWork = Str::is('*wxwork*', request()->userAgent());
```
Comment (atymic) [inline src/WxWork/Provider.php:44]: I assume this is detecting the mobile app?

File: src/WxWork/Provider.php [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+        return $this->credentialsResponseBody;
+    }
+
+    /**
+     * get access token.
+     *
+     * @return void
+     */
+    protected function getAccessToken()
```
Comment (atymic) [inline src/WxWork/Provider.php:139]: Why is this being cached?

File: src/WxWork/composer.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
+{
+    "name": "socialiteproviders/wxwork",
```
Comment (atymic) [inline src/WxWork/composer.json:2]: Please add to split script in the repo root (monorepo builder)

File: src/WxWork/Provider.php [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+namespace SocialiteProviders\WeixinWork;
+
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Str;
+use SocialiteProviders\Manager\Contracts\ConfigInterface;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
```
Comment (atymic) [inline src/WxWork/Provider.php:12]: Please add an `@see` docblock linking to the ouath docs

**Review by atymic — CHANGES_REQUESTED** [review]

File: src/WxWork/Provider.php (hunk 4 above)
Comment (keller31) [reply src/WxWork/Provider.php:44]: Used to check if it is running inside wxwork.

File: src/WxWork/Provider.php (hunk 5 above)
Comment (keller31) [reply src/WxWork/Provider.php:139]: AccessToken has a relatively long validity period,
Wxwork has limited frequency of obtaining AccessToken.

**Pull request closed without merging**